main.cpp

```diff
@@ -363,6 +416,38 @@ FloatType& FloatType::divide(float rhs)
     return *this;
 }
 
+FloatType& FloatType::pow(const FloatType& exp)
+{
+    powInternal(float(exp));
+    return *this;
+}
+
+FloatType& FloatType::pow(const DoubleType& exp)
+{
+    powInternal(static_cast<float>(double(exp)));
+    return *this;
+}
+
+FloatType& FloatType::pow(const IntType& exp)
+{
+    powInternal(static_cast<float>(int(exp)));
+    return *this;
+}
+
+FloatType& FloatType::pow(float exp)
+{
+    powInternal(exp);
```

Comment:
after you make `powInternal` chainable, revise these functions to return what `powInternal` returns.
- [ ] revise